File: rust/testdata/capabilities/probe.swift
import Foundation

protocol Renderable {
    func render() -> String
}

class Widget: BaseWidget, Renderable {
    var name: String = ""

    func render() -> String {
        return helper(name)
    }
}

func mainEntry() {
    let w = Widget()
    _ = w.render()
}

File: rust/testdata/capabilities/probe.rs
use crate::helper::helper;

pub trait Render {
    fn render(&self) -> String;
}

pub struct Widget {
    name: String,
}

impl Render for Widget {
    fn render(&self) -> String {
        helper(&self.name)
    }
}

pub fn main_entry() {
    let w = Widget { name: String::new() };
    w.render();
}
